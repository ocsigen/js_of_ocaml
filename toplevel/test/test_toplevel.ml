let () = Js_of_ocaml_toplevel.JsooTop.initialize ()

let fmt = Format.std_formatter

let () =
  Js_of_ocaml.Sys_js.set_channel_flusher stderr (fun str -> Printf.printf "<ERR>: %s" str)

let () =
  Js_of_ocaml_toplevel.JsooTop.execute
    true
    ~pp_code:fmt
    ~highlight_location:(fun _ -> ())
    fmt
    {|
let () = print_endline "hello";;
1+;;|}
