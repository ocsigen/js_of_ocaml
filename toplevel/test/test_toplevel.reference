
let () = print_endline "hello";;
hello
1+;;
<ERR>: Line 3, characters 2-4:
<ERR>: Error: Syntax error
