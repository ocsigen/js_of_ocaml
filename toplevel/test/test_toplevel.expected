
external parseInt : float -> int = "parseInt"
let f = 3.14
let () = Printf.printf "parseInt(%f) = %d\n" f (parseInt f);;
Dynlink: looking for symbol parseInt
parseInt(3.140000) = 3
external parseInt : float -> int = "parseInt"
val f : float = 3.14

let () = print_endline "hello";;
hello
1+;;
<ERR>: Line 3, characters 2-4:
<ERR>: Error: Syntax error

let () =
  let regex = Re.compile Re.(seq [str "//"; rep print ]) in
  Printf.printf "%b\n%!" (Re.execp regex "// a C comment");
  Printf.printf "%b\n%!" (Re.execp ~pos:1 regex "// a C comment");;
true
false
